Week2/HYF-JS3-HW2/index.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+{
+  function fetchJSON(url) {
+    return new Promise((resolve, reject) => {
+      const xhr = new XMLHttpRequest();
+      xhr.open('GET', url);
+      xhr.responseType = 'json';
+      xhr.onload = () => {
+        if (xhr.status < 400) {
+          resolve(xhr.response);
+        } else {
+          reject(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
+        }
+      };
+      xhr.onerror = () => reject(new Error('Network request failed'));
+      xhr.send();
+    });
+  }
+
+  function createAndAppend(name, parent, options = {}) {
+    const elem = document.createElement(name);
+    parent.appendChild(elem);
+    Object.keys(options).forEach(key => {
+      const value = options[key];
+      if (key === 'text') {
+        elem.textContent = value;
+      } else {
+        elem.setAttribute(key, value);
+      }
+    });
+    return elem;
+  }
+
+  const root = document.getElementById('root');
+  const header = createAndAppend('header', root, { class: 'header' });
+  createAndAppend('p', header, { text: 'HYF Repositories' });
+
+  const select = createAndAppend('select', header, { id: 'selector' });
+  const option = x => {
+    for (let i = 0; i < x.length; i++) {
+      createAndAppend('option', select, { text: x[i].name, value: i });
+    }
+  };
+
+  const mainContainer = createAndAppend('div', root, { id: 'container' });
+  const leftContainer = createAndAppend('div', mainContainer, { class: 'left-block frame' });
+  const table = createAndAppend('table', leftContainer);
+  const listContributor = createAndAppend('tbody', table);
+
+  const rightContainer = createAndAppend('div', mainContainer, { class: 'right-block frame' });
+  createAndAppend('p', rightContainer, {
+    text: 'Contributions',
+    class: 'contributor',
+  });
+
+  const list = createAndAppend('ul', rightContainer, { class: 'list1' });
+
+  function renderContributors(selRep) {
```

Comment:
You call this function with an array of contributors. The parameter `selRep` wrongly suggests that you are passing the selected repository.